v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compilation-job.ts

```diff
@@ -99,22 +103,19 @@ export class CompilationJobImplementation implements CompilationJob {
     // from other files (e.g. new Foo()), and it won't output its bytecode if
     // it's not asked for. This would prevent EDR from doing any runtime
     // analysis.
-    const defaultOutputSelection: CompilerInput["settings"]["outputSelection"] =
-      {
-        "*": {
-          "*": [
-            "abi",
-            "evm.bytecode",
-            "evm.deployedBytecode",
-            "evm.methodIdentifiers",
-            "metadata",
-          ],
-          "": ["ast"],
-        },
-      };
-
-    // TODO: Deep merge the user output selection with the default one
-    const outputSelection = defaultOutputSelection;
+    const outputSelection = await deepClone(settings.outputSelection ?? {});
+    outputSelection["*"] ??= {};
+    outputSelection["*"][""] ??= [];
+    outputSelection["*"]["*"] ??= [];
+
+    outputSelection["*"][""].push("ast");
+    outputSelection["*"]["*"].push(
```

Comment:
Does it matter that there are duplicates?

Reply:
We have opted to defer on this question. If duplicates are a problem we can add explicit removal code.